libs/calib/calib.c

```diff
@@ -250,168 +250,241 @@ static int calib_magironRead(FILE *file, calib_data_t *cal)
 }
 
 
-static void calib_motlinDefaults(calib_data_t *cal)
+/* Returns 0 if new value `val` was successfully saved to `cal` as `paramName` parameter. -1 otherwise */
+static int calib_accorthEnter(const char *paramName, calib_data_t *cal, float val)
 {
-	cal->params.motlin.motorEq[0][0] = 0.968600;
-	cal->params.motlin.motorEq[0][1] = 0.034796;
+	matrix_t *mat = NULL;
+	unsigned int row, col;
+	float *slot;
 
-	cal->params.motlin.motorEq[1][0] = 1.031400;
-	cal->params.motlin.motorEq[1][1] = 0.085204;
+	if (strlen(paramName) != 3) {
+		return -1;
+	}
 
-	cal->params.motlin.motorEq[2][0] = 1.003427;
-	cal->params.motlin.motorEq[2][1] = 0.142546;
+	row = (uint8_t)(paramName[1] - '0'); /* convert character to unsigned int digit */
+	col = (uint8_t)(paramName[2] - '0'); /* convert character to unsigned int digit */
```

Comment:
`row` and `col` are unsigned int type.
```suggestion
	row = paramName[1] - '0'; /* convert character to unsigned int digit */
	col = paramName[2] - '0'; /* convert character to unsigned int digit */
```

Reply:
This has to do with MISRA rule 10.4 that does not allow for implicit casting of signed values to unsigned values and explicit casting should be used instead.